qiskit_quantuminspire/qi_jobs.py

```diff
@@ -1,16 +1,29 @@
-from datetime import datetime, timezone
+import asyncio
 from typing import Any, List, Union
 
-from compute_api_client import Result as JobResult
+from compute_api_client import ApiClient, PageResult, Result as RawJobResult, ResultsApi
 from qiskit.circuit import QuantumCircuit
 from qiskit.providers import JobV1 as Job
 from qiskit.providers.backend import Backend
 from qiskit.providers.jobstatus import JobStatus
 from qiskit.result.result import Result
 
+from qiskit_quantuminspire.api.client import config
+from qiskit_quantuminspire.api.pagination import PageReader
 from qiskit_quantuminspire.qi_results import QIResult
 
 
+class CircuitExecutionData:
+    """Class for bookkeping of individual jobs."""
+
+    def __init__(
+        self, circuit: QuantumCircuit, job_id: Union[int, None] = None, results: Union[List[RawJobResult], None] = None
+    ) -> None:
+        self.job_id = job_id
+        self.circuit = circuit
+        self.results = [] if results is None else results
```

Comment:
For now our code is built on the assumption that every job has a single result, so I think this should be reflected in this class (though it may change later should we implement hybrid stuff).

Reply:
changed